README.md

```diff
@@ -343,6 +343,31 @@ query param:
 > ./cloud-sql-proxy --unix-socket C:\cloudsql myproject:my-region:mysql
 > ```
 
+### Configuring Service Account Impersonation
+
+The Proxy supports [service account impersonation](https://cloud.google.com/iam/docs/impersonating-service-accounts).
+This allows the Proxy to act as a different service account, which can be useful
+for granting access to resources that are not accessible to the default IAM principal.
+
+To use service account impersonation, you must have the
+`iam.serviceAccounts.getAccessToken` permission on the IAM principal. You can grant this permission by assigning the
+`roles/iam.serviceAccountTokenCreator` role to the IAM principal.
+
+To impersonate a service account, use the `--impersonate-service-account` flag:
+
+> [!NOTE]: 
+>
+> The service account must have `Cloud SQL Instance User`, `Service Usage Consumer` and `Cloud SQL Client permissions`.
+> The `roles/iam.serviceAccountTokenCreator` role is not required on the impersonated service account but on the IAM principal impersonating the service account.
```

Comment:
> The proxy's IAM principal must have the `iam.serviceAccounts.getAccessToken` permission in order to impersonate another service account, usually by granting it the  `roles/iam.serviceAccountTokenCreator` role. 
> The impersonated service account must have the `Cloud SQL Instance User`, `Service Usage Consumer` and `Cloud SQL Client` permissions.

I've rearranged this for clarity: I put the statements in order and removed negatives. Service account impersonation is really difficult to explain.